gradle/javaProject.gradle

```diff
@@ -201,11 +201,21 @@ check {
     dependsOn createJavadoc
 }
 
+def getCheckedOutGitCommitHash() {
+    'git rev-parse --verify --short HEAD'.execute().text.trim()
+}
+
 publishing {
+    def artifactCodeCoverageReport = file("$buildDir/jacoco/test.exec")
     publications {
         mavenJava {
             from components.java
         }
+        if (artifactCodeCoverageReport.exists()) {
+            maven(MavenPublication) {
+                artifact source: artifactCodeCoverageReport, classifier: getCheckedOutGitCommitHash()
```

Comment:
It's better to append getCheckedOutGitCommitHash() to `version` or some other parameter and keep `classifier` as `jacoco`. I'll locally test this.

@ThisaruGuruge please note.

Reply:
Removed the commit hash from the classifier in 1f45bdf